<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="sequence_id"                         default="06" />
    <arg name="calib_file"                          default="$(find viso_pose)/param/kitti_odom.yaml"/>
    <arg name="cam0_topic"                          default="/cam0/image_raw"/>
    <arg name="cam1_topic"                          default="/cam1/image_raw"/>
    <arg name="cam0_info_topic"                          default="/cam0/camera_info"/>
    <arg name="cam1_info_topic"                          default="/cam1/camera_info"/>
    <arg name="odom0_topic"                          default="/cam0/odom"/>
    <arg name="data_dir" default="$(find viso_pose)/kitti-odom/dataset/sequences/$(arg sequence_id)" />
    <arg name="bags_dir" default="$(find viso_pose)/kitti-bags/" />
    <arg name="out_dir" default="$(find viso_pose)/kitti-bags/" />

    <param name="/use_sim_time" value="false" />
    <node pkg="viso_pose" type="kitti_data_node" name="kitti_data_node" output="screen"
          required="true"
          >
        <param name="data_dir" value="$(arg data_dir)" />
        <param name="delay" value="5" />
        <param name="rate" value="1" />
    </node>
    <!-- 
    <param name="/use_sim_time" value="true" />
    <node pkg="rosbag" type="play" name="kitti_data_node"
          args="-clock -d 5 -wait-for-subscribers $(arg bags_dir)/kitti_data_odometry_color_sequence_06.bag">
      <remap from="/kitti/camera_color_left/image_rect" to="$(arg cam0_topic)" />
      <remap from="/kitti/camera_color_left/camera_info" to="$(arg cam0_info_topic)" />
      <remap from="/kitti/camera_color_right/image_rect" to="$(arg cam1_topic)" />
      <remap from="/kitti/camera_color_right/camera_info" to="$(arg cam1_info_topic)" />
    </node>
    -->

    <node pkg="viso_pose" type="viso_node" name="viso_node" output="screen">
        <param name="viso_px_sigma" value="4" />
        <param name="cam0_topic" value="$(arg cam0_topic)" />
        <param name="cam1_topic" value="$(arg cam1_topic)" />
        <rosparam file="$(arg calib_file)" />
        <remap from="/viso_pose/relative_pose" to="$(arg odom0_topic)" />
        <remap from="/cam0/camera_info" to="$(arg cam0_info_topic)" />
        <remap from="/cam1/camera_info" to="$(arg cam1_info_topic)" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find viso_pose)/rviz/kitti_raw.rviz" />
    <node pkg="rosbag" type="record" name="record"
          args="$(eval ' '.join([
                arg('cam0_topic'),
                arg('cam1_topic'),
                arg('cam0_info_topic'),
                arg('cam1_info_topic'),
                arg('odom0_topic'),
                '-O',
                arg('out_dir') + '/kitti_libviso_odom_06.bag'
                ]))" />

</launch>
